src/wp-admin/admin.php

```diff
@@ -417,4 +417,10 @@
 	 * @since 2.6.0
 	 */
 	do_action( "admin_action_{$action}" );
+} else {
+	// Checks if the request URI is /wp-admin/admin.php without any query parameters
+	if ( '/wp-admin/admin.php' === $_SERVER['REQUEST_URI'] && empty( $_SERVER['QUERY_STRING'] ) ) {
+		wp_safe_redirect( admin_url() );
+		exit;
+	}
```

Comment:
```suggestion
} elseif ( '/wp-admin/admin.php' === $_SERVER['REQUEST_URI'] && empty( $_SERVER['QUERY_STRING'] ) ) {
	// Checks if the request URI is /wp-admin/admin.php without any query parameters
	wp_safe_redirect( admin_url() );
	exit;
```

This just removes the nesting in favor of using an elseif.

As WordPress can be installed in a sub-directory, the `REQUEST_URI` check will need to account for that. Possibly use `wp_parse_url( admin_url( '/admin.php' ), PHP_URL_PATH )` in place of the hard coded string.